### PR TITLE
Use device_id instead of config entry id and node id for zwave_js

### LINKED ISSUE #12136: Z-Wave options do not appear if Switch as X is first in Integration list
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

When I have a Z-Wave device setup with Switch as X, if it is listed first in the integrations the additional Z-Wave options do not show up in the device page. If Z-Wave JS shows up first I do get the proper config options.

image

### Describe the behavior you expected

Z-Wave configuration options should always be available:

image
image

### Steps to reproduce the issue

Configure multiple Z-Wave switch entities with "Switch as X" and view device page.

### What version of Home Assistant Core has the issue?

dev

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

all

### Which operating system are you using to run this browser?

Windows 11, iOS

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### LINKED ISSUE #12641: Switch as X integration disables zwave_js configuration
### The problem

I 'converted' nearly all of my Zwave switches to either lights or fans using the 'Switch as X' integration.  However, doing so means that I cannot write a new automation via the UI that triggers on state change of the device (e.g. it being turned on or off, or a zwave value changing).  Also once the switch is converted, you cannot view the zwave configuration for it anymore.

e.g. config of switch before - note the zwave config options

image

after:  zwave info not available

image

Trigger not available when switch as light

image

### What version of Home Assistant Core has the issue?

core-2022.5.1

### What was the last working version of Home Assistant Core?

_No response_

### What type of installation are you running?

Home Assistant OS

### Integration causing the issue

Switch as X

### Link to integration documentation on our website

https://www.home-assistant.io/integrations/switch_as_x

### Diagnostics information

_No response_

### Example YAML snippet

_No response_

### Anything in the logs that might be useful for us?

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by raman325**
## Proposed change
Instead of using config entry ID and node ID for any device specific WS commands, we now use device_id. This replaces #12642

Fixes https://github.com/home-assistant/frontend/issues/12641
Fixes https://github.com/home-assistant/frontend/issues/12136
Dependent on https://github.com/home-assistant/core/pull/71667

Primary changes:
- Got rid of dependencies on config entry ID and node ID where possible so we can just pass the device ID into the WS commands which is already available on these pages
- Updated the network_status WS API command to return the node statuses instead of a list of node IDs so that we can save an API call

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- Link to core pull request: https://github.com/home-assistant/core/pull/71667

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by raman325** [comment]
The related core PR has been approved

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-actions-zwave_js.ts [hunk 1]
```diff
@@ -96,33 +97,30 @@ export class HaDeviceActionsZWaveJS extends LitElement {
...
 
   private async _healNodeClicked() {
-    if (!this._nodeId || !this._entryId) {
+    if (!this.device) {
       return;
     }
     showZWaveJSHealNodeDialog(this, {
-      entry_id: this._entryId,
-      node_id: this._nodeId,
+      entry_id: this._entryId!,
```
Comment (bramkragten) [inline src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-actions-zwave_js.ts:113]: Do we still need `entry_id` here, can't we also make that work with device id?

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-info-zwave_js.ts [hunk 2]
```diff
@@ -90,11 +67,11 @@ export class HaDeviceInfoZWaveJS extends LitElement {
       <h4>
         ${this.hass.localize("ui.panel.config.zwave_js.device_info.zwave_info")}
       </h4>
-      ${this._multipleConfigEntries
```
Comment (bramkragten) [inline src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-info-zwave_js.ts:93]: The idea of this was that we only show the source if you have multiple networks.

**Review by bramkragten — DISMISSED** [review]

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-info-zwave_js.ts (hunk 2 above)
Comment (raman325) [reply src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-info-zwave_js.ts:93]: ah, didn't look at this closely enough. Reverted

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-actions-zwave_js.ts [hunk 3]
```diff
@@ -29,34 +28,36 @@ export class HaDeviceActionsZWaveJS extends LitElement {
 
   @state() private _entryId?: string;
 
-  @state() private _nodeId?: number;
-
   @state() private _node?: ZWaveJSNodeStatus;
 
-  protected updated(changedProperties: PropertyValues) {
+  public willUpdate(changedProperties: PropertyValues) {
     if (changedProperties.has("device")) {
```
Comment (balloob) [inline src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-actions-zwave_js.ts:34]: Always call super when overriding.
```suggestion
    super.willUpdate(changedProperties);
    if (changedProperties.has("device")) {
```

File: src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-info-zwave_js.ts [hunk 4]
```diff
@@ -29,57 +27,40 @@ export class HaDeviceInfoZWaveJS extends LitElement {
...
 
   @state() private _multipleConfigEntries = false;
 
-  @state() private _nodeId?: number;
-
   @state() private _node?: ZWaveJSNodeStatus;
 
-  protected updated(changedProperties: PropertyValues) {
+  public willUpdate(changedProperties: PropertyValues) {
     if (changedProperties.has("device")) {
```
Comment (balloob) [inline src/panels/config/devices/device-detail/integration-elements/zwave_js/ha-device-info-zwave_js.ts:37]:
```suggestion
    super.willUpdate(changedProperties);
    if (changedProperties.has("device")) {
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**